modules/kind/kindcluster.go

```diff
@@ -71,8 +71,43 @@ func (c *Cluster) ID() string {
 	return fmt.Sprintf("pkg.package-operator.run/cardboard/modules/kind.Cluster{name:%s}", c.name)
 }
 
+func (c *Cluster) IPv4() (string, error) {
+	clusterNodes, err := c.Nodes(false)
+	if err != nil {
+		// TODO: augment error
+		return "", err
+	}
+	for _, node := range clusterNodes {
+		role, err := node.Role()
+		if err != nil {
+			// TODO: augment error
+			return "", err
+		}
+		if role == "control-plane" {
+			ipv4, _, err := node.IP()
+			if err != nil {
+				// TODO: augment error
+				return "", err
+			}
+			return ipv4, nil
+		}
+	}
+	return "", fmt.Errorf("can't find control plane node for cluster %s", c.Name())
+}
+
 func (c *Cluster) Name() string { return c.name }
 
+func (c *Cluster) Nodes(internal bool) ([]nodes.Node, error) {
```

Comment:
what are internal nodes? why is this choice exposed when it is only called with `false`? :thinking:

Reply:
it is called with `false` in Package Operator, which is the only place we use this so far, but if we want this to be a standalone module I believe we should give the option to choose between internal nodes or not (since the provider gives this opportunity)

Reply:
What is the difference between an internal node and other nodes?